src/svs/inacademia_frontend.py

```diff
@@ -101,6 +103,16 @@ def _read_entity_id_map(self):
         with open(self.config['entity_id_map_path']) as f:
             return json.loads(f.read())
 
+    def _read_translation_id_map(self):
+        with open(self.config['translation_id_map_path']) as f:
+            return yaml.load(f, Loader=yaml.FullLoader)
+
+    def _translate_hint(self, hint):
+        result = self.translation_id_map.get(hint, None)
+        if result:
+            return self._translate_hint(result)
+        return hint
```

Comment:
I would write this with a simple `while` look, instead of recursion:

```suggestion
    def _get_fresh_hint(self, stale_hint):
        fresh_hint = stale_hint
        while fresh_hint in self.translation_id_map:
            fresh_hint = self.translation_id_map[fresh_hint]
        return fresh_hint
```

Note that I've renamed this to `_get_fresh_hint` and the variables are now the stale and the fresh hint. I think this naming is clearer with respect to what we're doing. Translation is the mechanism to achieve the effect, but the effect is about retrieving the _fresh_ hint, given a (possibly) _stale_ one.

Reply:
Will use your suggestion